src/tensorial_form.jl

```diff
@@ -86,16 +118,36 @@ A multi-dimensional array containing the coefficients.
 This implementation uses Julia's `Base.Cartesian` to generate a set of nested
 loops to compute the element ``A[i_1, …, i_n]``.
 """
+#FIXME: Correct output ordering in multi-array mode, incorrect 1D array mode.
 @generated function _loop_tensor!(A::AbstractArray{T, N}, v::VectorOfArray) where {T, N}
     quote
-    cache = Vector{Float64}($N-2)
-    @inbounds for $(Symbol(:i_, N)) in eachindex(v[$N])
-                  # FIXME: split me in several lines!
-                  @nloops $(N-1) i i -> eachindex(v[i]) d -> d == 1 ? nothing : d == $N-1 ? cache[d-1] = v[i_{d}, d] * v[i_{d+1}, d+1] : cache[d-1] = cache[d] * v[i_{d}, d] begin (@nref $N A i) = v[i_1, 1] * cache[1] end
-              end
-    end
+        @inbounds for $(Symbol(:i_, N)) in eachindex(v[$N])
+                  @nloops $(N-1) i i -> eachindex(v[i]) d -> d == 1 ?
+                            nothing :
+                            d == $N-1 ?
+                                    t_{d-1} = v[i_{d}, d] * v[i_{d+1}, d+1]:
+                                    t_{d-1} = t_{d} * v[i_{d}, d] begin
+                                        if $N==2
+                                            (@nref $N A i) = v[i_1, 1] * v[i_2, 2]
+                                        else
+                                            (@nref $N A i) = t_1 * v[i_1, 1]
+                                        end
+                                end
+               end
+     end
 end
 
+function _naive!(A::AbstractArray{T, N}, ncoeffs::Int64, v::VectorOfArray) where {T, N}
+
+    d = length(v)
+
+    @inbounds for i in 1:ncoeffs
+        mj = ind2sub(A,i)
+            @inbounds for j in 1:d
+                @views A[i] = A[i].*v[j][mj[j]]
```

Comment:
better use `v[mj[j], j]` with `VectorOfArray` so it does not create a temporary array

Reply:
Ok I thought it was just a different syntax since I used @views and dot operators